File: segundos 4 lenguajes/go/JugadorPro.go
package main

type JugadorPro struct {
	Nombre string
	Nivel  int
}

func NewJugadorPro() JugadorPro {
	return JugadorPro{"Juan", 10}
}

func NewJugadorProParams(nombre string, nivel int) JugadorPro {
	return JugadorPro{nombre, nivel}
}

func NewJugadorProNivel(nivel int) JugadorPro {
	return JugadorPro{"Desconocido", nivel}
}

func NewJugadorProCopy(j JugadorPro) JugadorPro {
	return JugadorPro{j.Nombre, j.Nivel}
}

File: segundos 4 lenguajes/Java/JugadorPro.java
package Java;
class JugadorPro {
    String nombre;
    int nivel;

    public JugadorPro() {
        this("Juan", 10);
    }

    public JugadorPro(String nombre, int nivel) {
        this.nombre = nombre;
        this.nivel = nivel;
    }

    public JugadorPro(int nivel) {
        this.nombre = "Desconocido";
        this.nivel = nivel;
    }

    public JugadorPro(JugadorPro p) {
        this.nombre = p.nombre;
        this.nivel = p.nivel;
    }
}

